### DISCUSSION TITLE
Не собирается проект 1.14

### DISCUSSION THREAD

**Discussion by Alekseiger426** [vvip-68/LedPanelWiFi#27] [Вопросы и ответы]
1.13 собрал все отлично работает, но не хватает управления через интернет. Хочется с улицы упралять режимами)
При компиляции получаю ошибку 
 case  1: /* D10/TX */ FastLED.addLeds<LED_CHIP,  1, COLOR_ORDER>(leds, led_start, led_count).setCorrection( TypicalLEDStrip ); break; // Для NodeMCU определены константы D9(3) и D10(1), но не определены RX И  TX; 
 Собираю на 8266, отладка отключена, device type -0. Нет ничего кроме модуля и ленты.
 Нигде не нашел такой проблемы, версии ядра и библиотек как в инструкции

**Comment by vvip-68** [comment]
1. Версия 1.13 - для управления с улицы нужно включить и настроить MQTT - https://github.com/vvip-68/GyverPanelWiFi/wiki/Настройка-подключения-к-MQTT-серверу.

2. Указанная строчка в которой ошибка - ее в версии 1.13 нет вообще. Есть в 1.14. Вы привели только текст строчки в которой ошибка, но текст самой ошибки нет. Хрустальный шар сломался, ничего не показывает...

Версия 1.14 не имеет управления через MQTT. Некому сформулировать как это должно работать и написать приложение для телефона, поддерживающее управление через MQTT, поэтому в версии 1.14 оно было выброшено.

В 1.14 можно попробовать пробросить порты, если есть белый IP
Ну или поставить на телефон какой-нибудь remote-control, через него заходить на домашний комп, который подключен к локальной сети и уже удаленно с этого компа управлять гирляндой.

**Reply by Alekseiger426** [reply]
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino: In function 'void allocateLeds()':
utility:944:43: error: parse error in template argument list
utility:944:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:102: error: 'Если' was not declared in this scope
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:944:65: note: in expansion of macro 'COLOR_ORDER'
utility:945:43: error: parse error in template argument list
utility:945:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:945:65: note: in expansion of macro 'COLOR_ORDER'
utility:946:43: error: parse error in template argument list
utility:946:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:946:65: note: in expansion of macro 'COLOR_ORDER'
utility:947:43: error: parse error in template argument list
utility:947:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:947:65: note: in expansion of macro 'COLOR_ORDER'
utility:948:43: error: parse error in template argument list
utility:948:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:948:65: note: in expansion of macro 'COLOR_ORDER'
utility:949:43: error: parse error in template argument list
utility:949:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:949:65: note: in expansion of macro 'COLOR_ORDER'
utility:950:43: error: parse error in template argument list
utility:950:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:950:65: note: in expansion of macro 'COLOR_ORDER'
utility:951:43: error: parse error in template argument list
utility:951:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:951:65: note: in expansion of macro 'COLOR_ORDER'
utility:952:43: error: parse error in template argument list
utility:952:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:952:65: note: in expansion of macro 'COLOR_ORDER'
utility:953:43: error: parse error in template argument list
utility:953:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:953:65: note: in expansion of macro 'COLOR_ORDER'
utility:954:43: error: parse error in template argument list
utility:954:43: error: statement cannot resolve address of overloaded function
In file included from D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:153:
a_def_hard.h:163:111: error: expected ';' before 'цвет'
D:\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\utility.ino:954:65: note: in expansion of macro 'COLOR_ORDER'
exit status 1
parse error in template argument list

**Reply by Alekseiger426** [reply]
Если я правильно вас понял это полный текст ошибки.
MQTT отключил в 1.13 так как пишут что виснет через какое-то время.
1.14 кину через доменное имя кинетика и смогу управлять откуда угодно. Так с WLED делал, но этотакая тормозная штука это самый влед- шаг в сторону и все крашится. Ваш проект на голову выше и стабильнее. 
Отдельное спасибо за проделанную работу

**Reply by vvip-68** [reply]
Какая версия IDE стоит? Какая версия ядра и FastLED?

**Reply by Alekseiger426** [reply]
![image](https://github.com/vvip-68/LedPanelWiFi/assets/149966899/154ddc01-aaf1-4e27-b472-d7d98ddebbdf)
Ядро 3.1.2
![image](https://github.com/vvip-68/LedPanelWiFi/assets/149966899/24819d56-1d3f-4f3c-bddb-8966f1c5fcd9)
FastLED 3.5.0
IDE 1.8.19

**Reply by Alekseiger426** [reply]
![image](https://github.com/vvip-68/LedPanelWiFi/assets/149966899/2a481f7a-6326-4073-b4d3-d12f20dcc2cd)
версия платы ESP-12E Node NCU 1.0

**Reply by vvip-68** [reply]
Ну, не знаю. Я с такой ошибкой не сталкивался.
Попробуйте в a_def_hard.h в строке 163 где 
`#define COLOR_ORDER       GRB' все что дальше - комментарий - удалить.

**Reply by Alekseiger426** [reply]
Бинго!!!)) скомпилировалось, сейчас буду шить. СПАСИБО.

А еще мучает вопрос по 1.13, у меня на роутере можно создать доменное имя для любого устройства с web интерфейсом.
Оно вида dev.user.keenetic.pro и при переходе по этой ссылке я попадаю в веб интерфейс  вне локальной сети, так вот в приложение Gyver Panel или Player при вводе IP можно вписать dev.user.keenetic.pro, но не подключается к гирлянде. Есть смысл копать настройки роутера , порты и тд или приложение по умолчанию не понимает что я хочу?

**Reply by vvip-68** [reply]
Приложение не понимает. Оно умеет работать только с IP адресами. И 1.13 не имеет Web-интерфейса.

**Reply by Alekseiger426** [reply]
![image](https://github.com/vvip-68/LedPanelWiFi/assets/149966899/4899fad8-28e7-448f-ad11-3c6875aa8ebf)
Прошил но вижу вот такую страницу когда перехожу по IP, уф чувствую что я уже близко к финалу)

**Reply by vvip-68** [reply]
Теперь переходим в wiki проекта, находим ссылку FAQ и почти в самом конце читаем.
Заколебался уже объяснять нечитающим инструкции...

**Reply by Alekseiger426** [reply]
Спасибо что направили, почти сам дошел до этого)
Все настроил, картинки даже залились.
Из локальной сети отлично работает, но через интернет не загружается. Это фиаско)))
![image](https://github.com/vvip-68/LedPanelWiFi/assets/149966899/d77c0d66-083a-4249-a0a5-4f29ed0d3944)

**Reply by vvip-68** [reply]
То есть сама страница по 80 порту (HTTP)  загружается и работает. Это хорошо.
Далее загруженная страница открывает соединение с устройством по WebSocket.
Судя по картинке - именно веб-сокет соединение не устанавливается...
Есть ли возможность вот как в обычном браузере в режиме разработчика F12 посмотреть что там по сетевому обмену, какие именно ошибки идут? Скриншот судя по размееру/формату - с экрана компьютера, значит в браузере можно нажать F12 перейти на вкладку  Сеть и посмотреть что там...

По идее WebSocket тоже работает over HTTP и все на том же 80 порту, но, наверное, есть какие-то нюансы...
Может быть при пробросе портов нужны какие-то дополнительные настройки

**Reply by Alekseiger426** [reply]
Да все так как вы и описали судя по ошибке, пойде читать на форуме keenetic
![image](https://github.com/vvip-68/LedPanelWiFi/assets/149966899/8f37d35a-9e88-435e-b6f8-20587773e2a1)

**Reply by vvip-68** [reply]
Увы... браузер (или роутер?)  не дает установить соединение по незащищенному каналу...
Нужно создавать защищенный сокет, но с этим проблема:
1. Не уверен, что библиотеки вебсокета, которые используются в скетче это умеют - надо изучать
2. Если и умеет - шифрование все-таки вещь ресурсоемкая - хватит ли мощи у ESP8266? Может быть ESP32 и смог бы, но... Тоже надо изучать.
3. По простому то соединению у ESP8266 памяти остается с гулькин нос. Если туда еще сертификаты и шифрование прикручивать... Не уверен что такое возможно

Ну и не изучал плотно я этот вопрос, поскольку проект изначально был ориентирован на работу в локальной сети...
Так что, видимо, облом-с...

Могу только предложить поставить на комп и телефон какой-нибудь remote desktop, подключаться с телефона к компу 
Я пробовал вот таким управлять 
![image](https://github.com/vvip-68/LedPanelWiFi/assets/39353139/50013d16-3a4d-449d-b83d-383ab8662e08)

![image](https://github.com/vvip-68/LedPanelWiFi/assets/39353139/3ebdd21a-8b8e-4892-ba88-749033266a7c)

Белых адресов не нужно. Работает через их сервер...

**Reply by huskydogsmile** [reply]
я пользуюсь "удаленный рабочий стол" от гугл и все без гемора

**Reply by vvip-68** [reply]
Что-то посмотрел - не впечатлило. Или нестабильно работает... Один раз удалось даже экран в горизонтальное положение на телефоне переключить. И потыкаться типа мышкой. Потом мышка пропала, экран телефона только вертикальный. И что бы не делал  - управлять удаленным компом уже не получилось. Или я чего-то не понимаю, или...

**Reply by Alekseiger426** [reply]
в итоге спасибо за помощь, буду экспериментировать с 1.13. На 1.14 почему-то начинает жутко тормозить через минут 10. Эффекты Синус, Лава и др сложные как слайд шоу. Причем удалил все картинки, оставил только  папку web. Попробую еще по MQTT посмотреть на 1.13

**Reply by vvip-68** [reply]
Интересно было бы еще увидеть что при этом в мониторе порта творится. Ну да ладно...

**Reply by Alekseiger426** [reply]
а я его отключил вроде, но попробую дождаться фризов и посмотреть. Пока не шил 1.13. Пробую другую 
настроить на ESP 32

**Reply by Alekseiger426** [reply]
Тишина в порту, при запуске неуий надор символов и все потом(

**Reply by vvip-68** [reply]
Ну, если вывод в порт отключен директивой в настройках компиляции, естественно туда ничего выводиться не будет...

**Reply by Alekseiger426** [reply]
23:04:13.429 -> WEB пакeт размером 10
23:04:13.429 -> <-- $6 7|WZ|E0
23:04:13.429 -> FM: 216980; Max: 113792
23:04:13.429 -> WEB пакeт размером 16
23:04:13.429 -> <-- $6 7|PW|FS|SX|EE
23:04:13.429 -> FM: 212652; Max: 113792
23:04:16.278 -> WEB пакeт размером 8
23:04:16.278 -> <-- $8 0 12;
23:04:16.278 -> FM: 217268; Max: 113792
23:04:16.278 -> Режим: Облака
23:04:24.732 -> WEB пакeт размером 8
23:04:24.732 -> <-- $8 0 33;
23:04:24.732 -> FM: 217072; Max: 113792
23:04:24.871 -> Режим: Синусы
23:04:39.897 -> Настройки сохранены в EEPROM
вот что в порту когда слайдшоу, причем это уже на плате esp32

**Reply by Alekseiger426** [reply]
:28:52.213 -> Режим: Звёзды
23:29:07.245 -> Режим: Узоры
23:29:22.240 -> Режим: Погода
23:29:27.736 -> dhcps: send_nak>>udp_sendto result 0
23:29:37.225 -> Режим: Анимация
23:29:48.107 -> dhcps: send_nak>>udp_sendto result 0
23:29:52.224 -> Режим: Камин
23:30:07.330 -> Режим: Стрелки
23:30:22.462 -> Режим: Дождь
23:30:37.515 -> Режим: Вышиванка
23:30:39.840 -> dhcps: send_nak>>udp_sendto result 0
23:30:52.538 -> Режим: Синусы
23:31:07.641 -> Режим: Спектрум
23:31:13.438 -> dhcps: send_nak>>udp_sendto result 0

или вот если не трогать и оно само листает эффекты и кстать web отваливается в этот момент

**Reply by vvip-68** [reply]
По логу (особенно в первой части) криминального ничего не вижу...
По второй части... А у вас задан конкретный IP адрес для контроллера или стоит 0.0.0.0, чтобы по DHCP получался?

У себя я такого не вижу - ни фризов ни вот этих сообщений...
По тексту dhcps: send_nak>>udp_sendto result 0 - похоже что контроллер пытается в сеть отправить подтверждение dhcp-серверу и у него это не получается. Это все сообщения из ядра, видимо из какого-то модуля, связанного с сетью. 

Предполагаю, что ядро начинает заниматься чем-то своим и не отдает управление скетчу, чтобы тот формировал картинку на матрицу. Отсюда и фризы...

Более ничего внятного сказать не могу, увы. Поставлю у себя на прогон, буду наблюдать...
Видимо, если хочется к новому году рабочую гирлянду, стоит найти-установить какой-нибудь другой проект...

В качестве бреда и если есть желание поэкспериментировать. Под ESP32 последнее ядро - 2.0.14
У меня проект под него собрался без ошибок, но сборку я не запускал, не проверял, отложил на потом...
Вдруг под ним контроллер будет как-то по другому работать. Вечером попробую собрать скетч на нем, посмотреть как работает и работает ли вообще...

**Reply by Alekseiger426** [reply]
Не, проект ваш буду использовать)) но 1.13 уже настроил. Сложилось такое ощущуение что:
1 ESP во время фризов начинает спамить что-то в сеть и падает интернет , в роутер не зайти, телевизор перестает работать(IP-TV), да в принципе  падает интренет весь. ИЛИ
2 со вчерашнего вечера мой провайдер сломался  и наоборот из-за проблем в сети ESP многократно пытается подключиться к интернту, возможно погоде или времени(хотя я их отключал) и не хватает ей времени на сложные эффекты. 
Кстати 1.13 всю ночь отработала и фпс не упал

**Reply by Alekseiger426** [reply]
По второй части... А у вас задан конкретный IP адрес для контроллера или стоит 0.0.0.0, чтобы по DHCP получался?
У меня забит 192.168.1.125 и в роутере на устройстве статический такой-же

**Reply by vvip-68** [reply]
Я погонял на версии ядра 2.0.14 - вроде работает. Версию выложил в репозиторий.
Неожиданно бегущая строка на 2.0.14 стала двигаться в два раза медленней примерно чем на 1.0.6 безо всякого другого изменения кода... Скорость движения в целом зависит от количества диодов. Я для ширины более 28 колонок сделал смещение сразу на две колонки вместо одной. Строка соответственно стала двигаться быстрее, но визуально, конечно, менее плавно...

Если есть лишний контроллер, чтобы не убивать уже настроенный 1.13 - можете на нем погонять 1.14 с новыми изменениями...

**Reply by Alekseiger426** [reply]
есть 8266 пару, мне нужно родителям собрать на выходных. Или скетч под ESP32?

**Reply by vvip-68** [reply]
Скетч и под то и под то. Речь то шла, что у вас на ESP32 проблемы и зависания. Я подумал может из за ядра. Потестировал на 2.0.14 - работает

На 8266 - мало памяти. Без активного Web-интерфейса остается примерно 17 кб памяти. И все вроде как крутится.
Если загрузить Web-страничку в браузер - выделяется дополнительная память под буферы самого Web-сервера, под строки обмена данными и так далее. В результате свободными остается примерно 6 Кбайт.  Этого уже маловато.

Я перебрал последнюю версию - попробовал как-то реорганизовать менеджмент памяти, чтобы не было фрагментации хипа. Когда вроде память свободная есть, но разбита на мелкие кусочки и выделить один затребованный большой не получается. И контроллер падает. Перезагружается.

Вот последние изменения были как раз направлены на попытку минимизировать фрагментацию и падения контроллера.
Так что с 8266 - всё еще довольно рискованная затея.  Я гонял на 8266 с 512 светодиодами - это для них такие размеры памяти остаются. Если у вас на ней матрица меньшего количества диодов - это лучше - больше свободной памяти останется.

Какие матрицы для родителей планируется собирать?

**Reply by Alekseiger426** [reply]
там до 400 диодов будет, 14х28 сетка по задумке или две 14х14. У меня как раз такая 14х14 сейчас на два окна.  Но я попробую новый скетч все равно и отпишусь

**Reply by SergeyBichan** [reply]
Не нашёл ничего похожего, поэтому напишу здесь. имеется ESP32 wroom 32 devkit. 4мб. 2 дня убил на подбор настроек. столкнулся с тем, что на версии FastLed 3.9.4 и ядре для ESP 3 там какой то из дискрипшена в самой прошивке, прошилось, завелось, через 5 минут начинает жёстко морозить эффекты, похоже не утечки некие памяти, но побоялся в работающий контроллер от бп пихать провод от ноута, чтобы глянуть чо там по логам(монитор порта). в итоге, версия ядра 2.017, фастлед 3.6.0. библиотеки само собой из архива с проектом влил. 2 дня работает очень круто. Спасибо вам @vvip-68, за вашу работу) прикрутить бы ещё, конечно, брокер сообщений, было бы супер. но по поводу брокера я тоже читал ваши сообщения) найдётся наверняка ктонибудь, кто расскажет и напишет для мобилы приложение.
А! и ещё! в лог постоянно сыпет про то что очередь переполнена сообщений. когда пытаешься с фронта подключиться после старта контроллера
15:46:49.981 -> <-- $6 7|W|H|HN|CRLE|PS|LG|VR|FS|SX|CH|CV|WZ|AL|SM|UP|FM|PN|MC|MX|MZ|TM
**15:46:49.981 -> FM: 179276  Max: 110580**
15:46:49.981 -> WEB пакeт размером 50
15:46:49.981 -> <-- $6 7|E1|E2|E3|E4|PS|BR|DM|RM|FV|PD|IT|EF|EN|FK|PZ2
15:46:49.981 -> FM: 180108  Max: 110580
15:46:50.012 -> ERROR: Too many messages queued
15:46:50.012 -> ERROR: Too many messages queued
15:46:50.012 -> ERROR: Too many messages queued

**Reply by vvip-68** [reply]
По сообщениям об To many messages queued. Там реально идет много запросов от интерфейса, чтобы получить информацию, необходимую для работы, особенно когда запуск в первый раз и все что можно еще не закэшировано. Но запросы повторяются, пока вся нужная информация не будет получена.

Утечки там быть не должно - я долго гонял с выводом в лог информации о свободной памяти. Все более-менее крутится возле какого-то одного размера занятой / свободной памяти и со временем не растет.

Тормоза - похоже на то, как идет запрос например данных о погоде с сервера и сервер не отвечает.
На время ожидания ответа контроллер останавливается в ожидании. Такая архитектура.

По поводу брокера - да, так никто и не собрался сказать, что конкретно ему требуется. В лучшем случае - ну чтобы к Алисе подключить можно было, а как оно - я не знаю. Ну вот и я не знаю. Пока не найдется человек с конкретным описанием что нужно - и как это должно работать - ничего и не будет.

Лично мне это не очень то и нужно для жизни. И чем далее во времене - тем хуже я уже помню что там в проекте как работает и тем менее реально что-то там поменять.

**Reply by vvip-68** [reply]
И еще... А дайте если можно лог после старта котроллера. Там версии библиотек отображаются. Еще Too many messages бывает, если берут библиотеку веб-сервера не той версии или не из проекта...

**Reply by SergeyBichan** [reply]
ок, вечером дома подкину к компу, покажу. насчёт вебсервера.. чтото я с ним мутил. не могу вспомнить, столько вариаций пробовал чтобы собрать наконец, что уже запутался. Вечером скину!) спасибо) каждый декабрь одно и тоже)) долбилово с ide)))

**Reply by SergeyBichan** [reply]
НЕ! всё же библиотека та. брал из проекта, забыл что на работе то я его и шил. посмотрел в ардуино IDE. 
<img width="305" height="197" alt="image" src="https://github.com/user-attachments/assets/5cb100ec-ead8-4161-80f0-538271e57653" />

**Reply by vvip-68** [reply]
А это - ни о чем не говорит. Говорит информация, выводимая в логе в мониторе порта.
Хотя... Все же говорит. Версия в комплекте проекта (с исправлениями) - 1.2.3
А у вас тут какая-то леваля 1.2.4 судя по всему установлена...

**Reply by vvip-68** [reply]
Опять нет --- 1.2.3 это вроде для 8266, для esp32 - все же 1.2.4. Но нужен лог... Там все видно будет...

**Reply by vvip-68** [reply]
Ну, или дайте первые строки после комментариев в файле ESPAsyncWebServer.h библиотеки,
чтобы до вечера до дома не ждать

**Reply by SergeyBichan** [reply]
> Опять нет --- 1.2.3 это вроде для 8266, для esp32 - все же 1.2.4. Но нужен лог... Там все видно будет...

Именно так. 1.2.4 для есп32, 1.2.3 для усп8266, ставил и ту и ту) всё пробовал уже) ща гляну чо там по строкам в самой библиотеке

**Reply by SergeyBichan** [reply]
<img width="911" height="970" alt="image" src="https://github.com/user-attachments/assets/72801cbd-ccf9-41ae-aaaa-1fc7b49459b7" />
вот скрин, если это нужно было

**Reply by SergeyBichan** [reply]
а не, стоп, вы скзаали после коментов
<img width="1165" height="713" alt="image" src="https://github.com/user-attachments/assets/6ac5e5a6-de04-47c5-9d82-44fee324c044" />

**Reply by vvip-68** [reply]
Судя по скриншотам - вроде правильная библиотека...

**Reply by SergeyBichan** [reply]
ну тогда вечером будут логи.

**Reply by SergeyBichan** [reply]
и то, это если вам самому интересно. если нет, то я не буду доставать) я думал может вдруг сталкивались) я понимаю, что это по сути пет проект ваш, если ещё такие как мы будем тыкать куда то в код, то не правильно это)

**Reply by vvip-68** [reply]
Если используется именно ЭТА версия библиотеки - логов не нужно.

**Reply by SergeyBichan** [reply]
да, это именно здесь я заливал прошивку на контроллер. с этого компа, имею ввиду. и только эта библиотека установлена.

**Reply by SergeyBichan** [reply]
и да, сценарий такой. пока в первый раз не конектишься на вебморду, этих сообщений нет. как тока туда лезешь, начинает ругаться) но проходит и потом без этого всё дальше идёт. чот пришёл я, суеты навёл и ушёл. вечером ещё потыкаю, посмотрю. если что отпишусь. Спасибо)

**Reply by vvip-68** [reply]
По задумке, браузер загружает все необходимое для работы себе в кэш и хранит там. Далее проверяется контрольная сумма данных и если она не изменилась - данные заново не посылаются, берутся из кэша. Но пока кэш не накоплен и не сохранен - все качается с маломощного микроконтроллера.

**Reply by SergeyBichan** [reply]
в общем, вопросов на балансе больше не имеется. всё работает корректно. гонял эти дни все. всё ок. СПАСИБО вам за вашу работу)
